pkgs/development/perl-modules/Parse-Win32Registry/default.nix

```diff
@@ -0,0 +1,40 @@
+{ lib
+, perlPackages
+, fetchurl
+, perl
+}:
+
+perlPackages.buildPerlPackage rec {
+  pname = "Parse-Win32Registry";
+  version = "1.1";
+
+  src = fetchurl {
+    url = "https://cpan.metacpan.org/authors/id/J/JM/JMACFARLA/Parse-Win32Registry-${version}.tar.gz";
+    hash = "sha256-wWOyAr5q17WPSEZJT/crjJqXloPKmU5DgOmsZWTcBbo=";
+  };
+
+  buildInputs = [ perl ];
+
+  preBuild = ''
+    unset PERL5LIB PERL_MM_OPT PERL_LOCAL_LIB_ROOT
+    export PERL_MM_USE_DEFAULT=1 PERL_AUTOINSTALL=--skipdeps \
+      PERL_MM_OPT="INSTALLDIRS=vendor DESTDIR='$pkgdir'" \
+      PERL_MB_OPT="--installdirs vendor --destdir '$pkgdir'" \
+      MODULEBUILDRC=/dev/null
+  '';
+
+  preInstall = ''
+    unset INSTALLDIRS DESTDIR PERL5LIB PERL_MM_OPT PERL_LOCAL_LIB_ROOT
+  '';
+
+  enableParallelBuilding = true;
```

Comment:
Are those `unset` and `export` applied if we use `buildPerlPackage`?

Reply:
They're generally set in https://github.com/NixOS/nixpkgs/blob/97f2135e6a6a3b85e323e3ede61fb9cec134ba93/pkgs/development/perl-modules/generic/builder.sh#L26 but in this case of a plain pure-Perl module, it makes little to no sense to override (as they'll be unset/exported by upstream ExtUtils::MakeMaker anyway,) and we also try to use `site` rather than `vendor` [installprefix](https://metacpan.org/dist/perl/view/INSTALL#installprefix).

Reply:
Fixed